### PR TITLE
Voronoi #30

### PR DESCRIPTION
**Pull request by virgile-baudrot**
Add Discrete Voronoi generation using NearestNeighborElement #30 

**Type of change**   
Please indicate the relevant option(s)

- [ ] :bug: Bug fix (non-breaking change which fixes an issue)
- [x] :sparkle: New feature (non-breaking change which adds functionality)
- [ ] :boom: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] :book: This change requires a documentation update

**Checklist**
- [x] The changes are documented
  - [x] The docstrings of the different functions describe the arguments, purpose, and behavior 
  - [x] There are examples in the documentation website
- [x] The changes are tested
- [x] The changes **do not** modify the behavior of the previously existing functions
  - If they **do**, please explain why and how in the introduction paragraph
- [ ] For **new contributors** - my name and information are added to `.zenodo.json`
- [ ] The `Project.toml` field `version` has been updated
  - Change the *last* number for a `v0.0.x` series release, a bug fix, or a performance improvement
  - Change the *middle* number for additional features that *do not* break the current test suite (unless you fix a bug in the test suite)
  - Change the *first* number for changes that break the current test suite

### CODE REVIEW

File: src/algorithms/discretevoronoi.jl [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    n::Int64
+    function DiscreteVoronoi(n::Int64)
+        @assert n > 0
+        new(n)
+    end
+end
+
+DiscreteVoronoi() = DiscreteVoronoi(3)
+
+function _landscape!(mat, alg::DiscreteVoronoi)
```
Comment (mkborregaard) [inline src/algorithms/discretevoronoi.jl]: Couldn't this whole function just be 
```julia
_landscape!(mat, alg::DiscreteVoronoi) = _landscape!(mat, NearestNeighborElement(alg.n, 1))
```
end of story?

File: src/algorithms/discretevoronoi.jl [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+"""
+struct DiscreteVoronoi <: NeutralLandscapeMaker
+    n::Int64
+    function DiscreteVoronoi(n::Int64)
+        @assert n > 0
+        new(n)
+    end
+end
+
+DiscreteVoronoi() = DiscreteVoronoi(3)
```
Comment (mkborregaard) [inline src/algorithms/discretevoronoi.jl]: why not just add 3 as a default value in the internal constructor in line 9?

File: src/algorithms/discretevoronoi.jl [hunk 3]
```diff
@@ -0,0 +1,32 @@
+"""
+    DiscreteVoronoi
+
+Assigns a value to each patch using a 1-NN algorithmm with `n` initial clusters
```
Comment (mkborregaard) [inline src/algorithms/discretevoronoi.jl]: This doesn't explain why it's called "Voronoi"

**Review by mkborregaard — CHANGES_REQUESTED** [review]
I'm not really seeing why it's necessary to reimplement here?

File: src/algorithms/discretevoronoi.jl (hunk 1 above)
Comment (virgile-baudrot) [reply src/algorithms/discretevoronoi.jl]: Indeed. We probably don't need to reimplement. Just add in documentation that `NearestNeighborElement` with `k` set at 1 is voronoi-like.

Comment (virgile-baudrot) [reply src/algorithms/discretevoronoi.jl]: Basically, I was just looking for a way to rasterize a Voronoi (#30). Now I know it's super easy using `NearestNeighborElement`, we can close the issue 👍

Comment (mkborregaard) [reply src/algorithms/discretevoronoi.jl]: Ah no, I think it's great to have a wrapper for this! i just wanted to reuse the function we already had. I think it should work if you edit the PR as suggested :-)

Comment (mkborregaard) [reply src/algorithms/discretevoronoi.jl]: Sorry I wasn't clear what I meant by "reimplement"

File: src/algorithms/discretevoronoi.jl (hunk 2 above)
Comment (virgile-baudrot) [reply src/algorithms/discretevoronoi.jl]: oh nice. I see

**Review by mkborregaard — APPROVED** [review]
Nice @virgile-baudrot looks good - thanks for the contribution!

**Pull request merged**